### PR TITLE
*: refactor some decode functions

### LINKED ISSUE #31232: refine some decode function
## Enhancement
We can refine some decode functions.
1. We don't need to check NeedRestoredData, which depends on global variables `NewCollationEnabled`.
2. We should follow the `column, handle, default value, null` decode logic, but some decode function doesn't follow it.
3. GetOldrow function is too complex.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #31232
Problem Summary:
This PR tries to refactor some decode functions.
1. We don't need to check NeedRestoredData, which depends on global variables NewCollationEnabled.
2. We should follow the column, handle, default value, null decode logic, but some decode function doesn't follow it.
3. GetOldrow function is too complex.
### What is changed and how it works?
1. Rearrange the decode logic, and remove the check `NeedRestoredData` since it's not needed.
2. Refacore getOldRow and DecodeRawRowData to make them more readable.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: executor/point_get.go [hunk 1]
```diff
@@ -584,13 +586,10 @@ func tryDecodeFromHandle(tblInfo *model.TableInfo, schemaColIdx int, col *expres
 		chk.AppendInt64(schemaColIdx, handle.IntValue())
 		return true, nil
 	}
-	if types.NeedRestoredData(col.RetType) {
-		return false, nil
-	}
 	// Try to decode common handle.
 	if mysql.HasPriKeyFlag(col.RetType.Flag) {
 		for i, hid := range pkCols {
-			if col.ID == hid && notPKPrefixCol(hid, prefixColIDs) {
```
Comment (tangenta) [inline executor/point_get.go:593]: The argument `prefixColIDs` is not used anymore. We can remove it from the signature.

File: ddl/column_change_test.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ import (
 	"github.com/pingcap/tidb/util/testutil"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
+  "go.uber.org/zap"
```
Comment (tangenta) [inline ddl/column_change_test.go:42]: Please format this line.

**Review by xiongjiwei — APPROVED** [review]

**Pull request closed without merging**